elastic/eui
# [Storybook] Support storybook interactions for VRT

## Summary

This PR enables support for using storybook `interactions` for our VRT via Loki.

The concept is to use the newly added helper `lokiPlayDecorator` on a stories' `play` function in which specific component states based on user interaction can be simulated.

ℹ️ As we're currently not using any interactions for our components the interactions are by default only enabled when Loki is running via CLI.

### Changes

* ensures `@storybook/test` works as expected by resolving modules file
* adds a `lokiPlayDecorator` function - this helper ensures that interactions are included and working as expected in Loki VRT runs
* adds an example usage of `lokiPlayDecorator` for the "With Tooltip" story of `EuiComboBox` (ensures the tooltip is part of the VRT)
* updates VRT reference images for the "with Tooltip" story of `EuiComboBox`
* updates documentation on VRT to include new functionality

## QA

- [x]  checkout this PR locally `gh pr checkout 7735` and
 - got to `combo_box.stories.tsx`
 - update the `play` function body, e.g. hover a different option like:
  ```js
line 110: await userEvent.hover(options[0]); // change options[0] to options[1]
```
 - run `yarn test-visual-regression update --storiesFilter EuiComboBox`
 - confirm the VRT reference image updated, showing the 2nd option tooltip instead of one of the 1st option
- [x] review "With Tooltip" `EuiComboBox` [story](https://eui.elastic.co/pr_7735/storybook/index.html?path=/story/forms-euicombobox--with-tooltip) and confirm the interactions don't show in the "interactions" panel
- [x] review the updated [wiki page](https://github.com/elastic/eui/blob/020db34090461338c3bbbfe0c04a7ebe8257a1af/wiki/contributing-to-eui/testing/visual-regression-testing.md)
